adapters/pubmatic/pubmatic.go

```diff
@@ -655,3 +666,19 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 	}
 	return bidder, nil
 }
+
+// getDisplayManagerAndVer returns the display manager and version from the request.app.ext or request.app.prebid.ext source and version
+func getDisplayManagerAndVer(app *openrtb2.App) (string, string) {
+	if source, err := jsonparser.GetString(app.Ext, openrtb_ext.PrebidExtKey, "source"); err == nil && source != "" {
+		if version, err := jsonparser.GetString(app.Ext, openrtb_ext.PrebidExtKey, "version"); err == nil && version != "" {
+			return source, version
+		}
+	}
+
+	if source, err := jsonparser.GetString(app.Ext, "source"); err == nil && source != "" {
+		if version, err := jsonparser.GetString(app.Ext, "version"); err == nil && version != "" {
+			return source, version
+		}
+	}
```

Comment:
As mentioned earlier [here](https://github.com/prebid/prebid-server/pull/4000/files#r1809461812), you're missing JSON test coverage.
Please add another display manager JSON test where the source and version are present at `app.ext` but not at `app.ext.prebid` to cover these lines.

Reply:
Done